<launch>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="use_sim_time" value="true"/>
    <arg name="debug" value="false"/>
    <arg name="world_name" value="$(find ford_project)/worlds/course.world"/>
  </include>

  <include file="$(find ford_project)/launch/includes/kobuki.launch.xml">
    <arg name="base" value="kobuki"/>
    <arg name="stacks" value="hexagons"/>
    <arg name="3d_sensor" value="kinect"/>
    <arg name="init_pose" value="-x -1.0 -y -2.3"/>
  </include>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
    <param name="publish_frequency" type="double" value="30.0" />
  </node>

  <!-- Fake laser -->
  <node pkg="nodelet" type="nodelet" name="laserscan_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="depthimage_to_laserscan"
        args="load depthimage_to_laserscan/DepthImageToLaserScanNodelet laserscan_nodelet_manager">
    <param name="scan_height" value="10"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/camera/depth/image_raw"/>
    <remap from="scan" to="/scan"/>
  </node>
  
  <!-- nodes for IMU teleop -->
  <node pkg="ford_project" type="imu_manager" name="imu_manager"/>
  <node pkg="ford_project" type="imu_filter_Kalman.py" name="tilt_filter">
  	<param name="filter_mode" value="off"/>
  </node>
  <node pkg="ford_project" type="button_detect.py" name="button_event_detector"/>
  <node pkg="ford_project" type="imu_teleop.py" name="imu_teleop">
  	<param name="data_file_path" value="$(env HOME)/Desktop/data"/>
  </node>
  <node pkg="ford_project" type="velocity_smoother.py" name="velocity_smoother">
  	<param name="maximum_increment_linear" type="double" value="0.02"/>
  	<param name="maximum_increment_angular" type="double" value="1.0"/>
  	<param name="velocity_cap_linear" type="double" value="1.0"/>
  	<param name="velocity_cap_angular" type="double" value="3.0"/>
  	<remap from="cmd_vel_smooth" to="cmd_vel_mux/input/teleop"/>
  </node>
</launch>

